### DISCUSSION TITLE
Trigger workflow on push-tag event to a specific branch

### DISCUSSION THREAD

**Discussion by benipeled** [community/community#13226] [Actions]
We currently use the push-tag event as a trigger for the release pipeline as follows

```yaml
on:
  push:
    tags:
      - 'v**'
```
The problem is that it is not possible to filter a tag event pushed to a particular branch which means, every push-tag event **regardless of the branch**, trigger the release pipeline, which might be a bit messy in case of a tag pushed to dev or any other non-release branch,

AFAIK, from git perspective, the tag has nothing with the branch but it's a pointer to a particular commit regardless of the branch (which is another pointer to another commit) so it's probably not straightforward,

I know there is an option to trigger the pipeline in a Release event but I see no point in adding another (manually) layer to the pipeline when we anyway pushing a tag for a version.

**Comment by ethomson** [comment]
> AFAIK, from git perspective, the tag has nothing with the branch but it's a pointer to a particular commit regardless of the branch (which is another pointer to another commit) so it's probably not straightforward,

You're very much correct @benipeled - it's definitely not straightforward for that reason. I'll put this on the backlog, it's possible that there's something clever we can do here. I'll chat with some of the git teams here to see if they have any clever ideas.

Thanks much for taking the time to provide us with feedback.

**Reply by naviat** [reply]
@ethomson So, is there any updates for this backlog item?

**Comment by benipeled** [comment]
@ethomson thanks for the reply,
The tag<-->branch info can be gathered by `git branch --contains <tag-name>` which returns all branches containing the given tag.

**Reply by MarkStega** [reply]
I just ran into the same issue that @benipeled has,  In my case, I am running two branches, one for the 'current' code/releases (my 4.x versions) and one for 'LTS' code/releases (my 3.x versions). My workaround is to pattern match on the tag and only run the release pipeline for the branch if the tag is appropriate for the workflow.
```
on:
  push:
    tags:
      - '4.*' # Push events to matching pattern, i.e. 4.0.0, 4.1.0

**Reply by wpyoga** [reply]
@benipeled How do we use it to control whether the workflow or the jobs run though?

**Reply by TBG-FR** [reply]
Here's an example on how to use `git branch --contains <tag-name>` in a Github Action to execute a job only if a tag exists on a specific branch : https://stackoverflow.com/a/77727776/8179249 😉

**Comment by Zwyx** [comment]
Here's how I do it:

```yaml
on:
  push:
    tags:
      - "v[0-9]+.[0-9]+.[0-9]+"

jobs:
  build-website:
    if: github.event.base_ref == 'refs/heads/main'
    ...
```

With this, my workflow is triggered only when a **_lightweight_** tag is pushed and is placed on the last commit of the `main` branch.

> What is a lightweight catch? → Git has [two kinds of tags](https://git-scm.com/book/en/v2/Git-Basics-Tagging): _annotated_ and _lightweight_. This only works with lightweight tags.

Commands like `npm version` create annotated tags, so I just added a `postversion` script in my `package.json` to convert the newly created annotated tag to a lightweight one:

```bash
tag=$(git tag --points-at HEAD)

git tag -d "$tag"
git tag "$tag" HEAD
```

Full project here: https://github.com/Zwyx/library-of-babel
